internal/run/run.go

```diff
@@ -0,0 +1,310 @@
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/speakeasy-api/openapi-generation/v2/pkg/generate"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/internal/config"
+	"github.com/speakeasy-api/speakeasy/internal/download"
+	"github.com/speakeasy-api/speakeasy/internal/overlay"
+	"github.com/speakeasy-api/speakeasy/internal/sdkgen"
+	"github.com/speakeasy-api/speakeasy/internal/utils"
+	"github.com/speakeasy-api/speakeasy/internal/validation"
+	"github.com/speakeasy-api/speakeasy/pkg/merge"
+)
+
+func Run(ctx context.Context, target, source, genVersion, installationURL, repo, repoSubDir string, debug bool) error {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	wf, _, err := workflow.Load(workingDir)
+	if err != nil {
+		return err
+	}
+
+	if err := wf.Validate(generate.GetSupportedLanguages()); err != nil {
+		return err
+	}
+
+	targets := []string{}
+	for targetID := range wf.Targets {
+		targets = append(targets, targetID)
+	}
+	slices.Sort(targets)
+
+	sources := []string{}
+	for sourceID := range wf.Sources {
+		sources = append(sources, sourceID)
+	}
+	slices.Sort(sources)
+
+	if target == "" && source == "" {
+		if len(wf.Targets) == 1 {
+			target = targets[0]
+		} else if len(wf.Targets) == 0 && len(wf.Sources) == 1 {
+			source = sources[0]
+		} else {
+			// TODO update to use our proper interactive code
```

Comment:
I'll update this, either in this PR or my own, depending on which gets merged first